products/views.py

```diff
@@ -36,37 +36,45 @@ def get(self, request):
         try:
             type_id = request.GET.get("typeId", None)
             page = int(request.GET.get("page", 1))
-            
+            search_keyword = request.GET.get("keyword", None)
+
             limit = 28
-            offset = (page-1)*limit
+            offset = (page - 1) * limit
+
+            if search_keyword:
+                products = Product.objects.filter(name__icontains=search_keyword)
+
+                if not products.exists():
+                    return JsonResponse({"MESSAGE": "PRODUCT NOT FOUND"}, status=204)
+            else:
+                products = Product.objects.filter(type_id=type_id)
```

Comment:
지원님 엔드포인트 잘 합쳐 주셨습니다!

추가로 피드백 더 드리자면, 하나의 엔드포인트에서 filter 그리고 search 기능을 유연하게 구현하려면 Q객체를 사용해서 구현해보세요~ 그리고 dictionary를 이용해서도 필터 기능을 구현해보시고, 각 방법에 장점과 단점은 무엇인지 고민해보시면 좋은 공부가 될 것 같습니다!

```suggestion
           q = Q()
           
            if search_keyword:
                q &= Q(name__icontains=search_keyword)
            if type_id:
               q &= Q(type_id=type_id)

            products = Product.objects.filter(q)
```

Reply:
수정완료했습니다!